src/sentry/snuba/metrics/datasource.py

```diff
@@ -189,8 +194,9 @@ def _fetch_tags_or_values_per_ids(
     values available for those projects. In addition, when exactly one metric name is passed in
     metric_names, then the type (i.e. mapping to the entity) is also returned
     """
+    assert len({p.organization_id for p in projects}) == 1
     try:
-        metric_ids = _get_metrics_filter_ids(metric_names)
+        metric_ids = _get_metrics_filter_ids(projects, metric_names)
```

Comment:
nit: since we have many args now, lets be explicit 
```suggestion
        metric_ids = _get_metrics_filter_ids(projects=projects, metric_names=metric_names)
```